src/Plugin/ExtraField/Display/ListPageBaseFieldDisplay.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_list_pages\Plugin\ExtraField\Display;
+
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\extra_field\Plugin\ExtraFieldDisplayFormattedBase;
+use Drupal\oe_list_pages\ListBuilderInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Base field for list page extra fields.
+ */
+abstract class ListPageBaseFieldDisplay extends ExtraFieldDisplayFormattedBase implements ContainerFactoryPluginInterface {
```

Comment:
Class name should end with Base: ListPageExtraFieldBase.

And the doc above should not mention "base fields" cause those are something else :)